backend/src/main/controller/PromotionController.ts

```diff
@@ -89,14 +91,27 @@ export class PromotionController {
         promotionDTO.discount.discountType,
         promotionDTO.discount.discountValue
       );
+      const schedules = promotionDTO.schedules.map(
+        (scheduleDTO: ScheduleDTO) => {
+          return new Schedule(
+            scheduleDTO.startTime,
+            scheduleDTO.endTime,
+            scheduleDTO.dayOfWeek,
+            scheduleDTO.isRecurring
+          );
+        }
+      );
```

Comment:
One thing to note is that the incoming request body will validate **all** the discounts and schedules automatically as well, it's very powerful 😲 

This is just a couple of lines above, but note how I didn't need to explicitly mention, validate the schedule part of the `request.body` or validate the discount part of the `request.body`
https://github.com/ubclaunchpad/foodies/blob/b42162ce71da7c85e879017ac4d1f3aa98bcbbcc/backend/src/main/controller/PromotionController.ts#L82-L85

This is an example POST request, and all this is the format of a PromotionDTO. Also take a look at `PromotionValidation.ts` to see how we are validating `Discount`'s and `Schedules` as well if you are curious.

![image](https://user-images.githubusercontent.com/49849754/99868962-18fb6600-2b7c-11eb-9342-fce4c9cabe56.png)

Reply:
Is the time automatically PST/PDT?

Reply:
yup!